### PR TITLE
Fix: A modal window takes place when it is in a hidden state #1622 

### PR DESCRIPTION
**Pull request by govi2010**
This file changes is to fixed white space taken by model (clr-modal) when it is hidden.

So when model is hidden, I have applied display property none and when model is open i have added display property inline.

### CODE REVIEW

File: src/clarity-angular/modal/modal.ts [hunk 1]
```diff
@@ -26,7 +27,7 @@ import {GHOST_PAGE_ANIMATION} from "./utils/ghost-page-animations";
     viewProviders: [ScrollingService],
     templateUrl: "./modal.html",
     styles: [`
-        :host { display: inline-block; }
+        :host { display: inline; }
```
Comment (adityarb88) [inline src/clarity-angular/modal/modal.ts]: Why was this changed to `inline` from `inline-block`? I thought we just had to add `display: none` when the modal isn't open. I think thats what @mathisscott mentioned here: https://github.com/vmware/clarity/issues/1622#issuecomment-339349623

File: src/clarity-angular/modal/modal.ts [hunk 2]
```diff
@@ -106,8 +107,16 @@ export class Modal implements OnChanges, OnDestroy {
     ngOnChanges(changes: {[propName: string]: SimpleChange}): void {
         if (!this.bypassScrollService && changes && changes.hasOwnProperty("_open")) {
             if (changes._open.currentValue) {
+                const hostElement = this.elementRef.nativeElement;
```
Comment (adityarb88) [inline src/clarity-angular/modal/modal.ts]: Instead of getting the `ElementRef`, can we not just use a class which is applied to the host when the modal is not open? The class just has `display: none` on it and is removed when the modal is open.

File: src/clarity-angular/modal/modal.ts (hunk 1 above)
Comment (govi2010) [reply src/clarity-angular/modal/modal.ts]: @adityarb88  because when Popup is open. Due to `inline-block` margin can bee seen in background.

Like When you open modal  in backdrop you can see margin again. So I changed `inline-block ` to `inline`

File: src/clarity-angular/modal/modal.ts (hunk 2 above)
Comment (govi2010) [reply src/clarity-angular/modal/modal.ts]: I have removed `ElementRef ` and used  `@HostBinding("class.open")` so open class is added and added `:host { display: none; }  :host.open { display: inline; }` css.

File: src/clarity-angular/modal/modal.ts [hunk 3]
```diff
@@ -92,7 +94,7 @@ export class Modal implements OnChanges, OnDestroy {
     @Input("clrModalPreventClose") stopClose: boolean = false;
     @Output("clrModalAlternateClose") altClose: EventEmitter<boolean> = new EventEmitter<boolean>(false);
 
-    constructor(private _scrollingService: ScrollingService) {}
+    constructor(private _scrollingService: ScrollingService, private elementRef: ElementRef) {}
```
Comment (adityarb88) [inline src/clarity-angular/modal/modal.ts]: Thanks for making the changes. I believe we can remove the ElementRef now since its not being used.

**Comment by adityarb88** [comment]
@govi2010: Could you please squash your commits? I see 4 commits now but to merge this change in we need 1 commit only.

File: src/clarity-angular/modal/modal.ts [hunk 4]
```diff
@@ -6,7 +6,9 @@
 import {animate, AnimationEvent, state, style, transition, trigger} from "@angular/animations";
 import {
     Component,
+    ElementRef,
```
Comment (adityarb88) [inline src/clarity-angular/modal/modal.ts]: We can get rid of this import too.

Comment (govi2010) [reply src/clarity-angular/modal/modal.ts]: Yes sure.

**Comment by adityarb88** [comment]
@govi2010: Wanted to check if you could squash to commits so that we can merge this into master? Please let us know.

**Comment by govi2010** [comment]
@adityarb88  bit new to github. So i have messed up with some commits.  What changes you need from me ? I can do that right a way.

**Comment by adityarb88** [comment]
@govi2010: No worries. You can refer to this: https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit for more information. Let me know if that helps.

**Comment by adityarb88** [comment]
@govi2010 You could also submit a new PR if you are facing issues with this one. Its a small PR so I can review it again.

**Comment by govi2010** [comment]
@adityarb88 I am finding bit difficulty. But I think I need to learn one day , and it better to start from today.  I am running command blindly don't know what it is affect.

**Comment by adityarb88** [comment]
@govi2010: You can also refer to this video: https://www.youtube.com/watch?v=2E23I9PzplM. Let me know if that helps.
